In [4]:
import pandas as pd
import numpy as np

Project Goal

Analyze customer churn and identify factors associated with customer retention and attrition.

In [5]:
df=pd.read_csv('Churn_Modelling.csv')

Customer Churn Overview

Q1. What is the overall churn rate?

In [17]:
(df['Exited'].mean() *100 ).round(2)

20.37

The dataset reveals an overall churn rate of 20.37%, meaning that roughly one-fifth of the customer base has left the bank. Understanding the characteristics of these customers can help identify opportunities to improve retention and reduce future churn.

Q2. How many customers stayed and how many left?

In [7]:
df['Exited'].value_counts()

Exited
0    7963
1    2037
Name: count, dtype: int64

Demographic Analysis

Q3. Which country has the highest churn rate?

In [14]:
(df.groupby('Geography')['Exited'].mean()*100).sort_values(ascending=False).round(2)

Geography
Germany    32.44
Spain      16.67
France     16.15
Name: Exited, dtype: float64

Germany has the highest churn rate compared to other countries, indicating that customers from this region are more likely to leave the bank. In contrast, France and Spain show relatively lower churn rates, suggesting better customer retention in these markets.

Q4. Is churn higher among males or females?

In [9]:
df.groupby('Gender')['Exited'].mean()*100

Gender
Female    25.071539
Male      16.455928
Name: Exited, dtype: float64

The analysis shows that churn is higher among female customers compared to male customers. This suggests that female customers are more likely to exit the bank, indicating a potential need to investigate service experience, product fit, or engagement strategies targeted toward this segment.

Q5. Which age group has the highest churn?

In [10]:
df['AgeGroup'] = pd.cut(
    df['Age'],
    bins=[18,30,45,60,100],
    labels=['Young','Adult','Middle','Senior']
)
(df.groupby('AgeGroup' ,observed=False)['Exited'].mean()*100).round(2).sort_values(ascending=False)


AgeGroup
Middle    51.12
Senior    24.78
Adult     15.74
Young      7.50
Name: Exited, dtype: float64

Financial Analysis

Q6. What is the average balance of churned vs retained customers?

In [45]:
df.groupby('Exited')['Balance'].mean().round(2)

Exited
0    72745.30
1    91108.54
Name: Balance, dtype: float64

07.Do customers with higher salaries churn more?

In [47]:
df['SalaryGroup'] = pd.qcut(
    df['EstimatedSalary'],
    q=4,
    labels=['Low', 'Medium', 'High', 'Very High']
)
df.groupby('SalaryGroup',observed=False)['Exited'].mean() * 100

SalaryGroup
Low          20.00
Medium       19.80
High         20.12
Very High    21.56
Name: Exited, dtype: float64

Quantile-based binning (qcut) was applied to segment customers into equal-sized salary groups. This approach helps ensure fair comparison across income levels by avoiding imbalanced group sizes, which can occur with fixed interval binning (cut).

08.Does credit score affect churn?

In [11]:
df['Credit_band'] = pd.cut(
    df['CreditScore'],
    bins= [0,500, 600, 700, 800, 1000],
    labels=['Very Low','Low','Medium','High','Very High'])
(df.groupby('Credit_band' ,observed=False)['Exited'].mean()*100).round(2)


Credit_band
Very Low     23.64
Low          21.17
Medium       19.72
High         19.91
Very High    19.69
Name: Exited, dtype: float64

Customers with lower credit scores tend to have higher churn rates, suggesting that credit score may influence customer churn.

Customer Segmentation Analysis

Q9. Do active members churn less?

In [12]:
(df.groupby('IsActiveMember')['Exited'].mean() * 100).round(2)

IsActiveMember
0    26.85
1    14.27
Name: Exited, dtype: float64

Q10. Which geography has the highest average balance among churned customers only?

In [18]:
df[df['Exited'] == 1] \
  .groupby('Geography')['Balance'] \
  .mean() \
  .round(2) \
  .reset_index (name = 'avg balance') \
  .sort_values(by='avg balance',ascending=True)

,Geography,avg balance
0,France,71192.80
2,Spain,72513.35
1,Germany,120361.08


This analysis identifies which geographic region has the highest average balance among customers who churned, helping highlight potential revenue loss by region.